app/public/js/profile.js

```diff
@@ -4,41 +4,51 @@ profile.js is used in the profile.blade.php view.
 
 var regions = [];
 
-/**
-When select-all checkboxes are checked, check every other associated checkbox.
-*/
-function selectAllChanged() {
-	if ( $(this).is(':checked') ) {
-		var tab = $(this).closest('[role="tabpanel"]');
-		var questions = tab.find('.questions input[type="checkbox"]');
-		questions.prop('checked', true);
+function initSelectAllText() {
+	var isAllChecked = true;
+	$("div.category").each( iterateCategory );
+	selectAllToggle();
+	bindCheckboxes();
+
+	function iterateCategory(index,category) {
+		var NumberOfCheckboxes = $(category).find("div.questions input:checkbox").length;
+		var NumberOfcheckedCheckboxes = $(category).find("div.questions input:checked").length;
+		var isAllChecked = (NumberOfCheckboxes == NumberOfcheckedCheckboxes);
+		if(isAllChecked) {
+			$(category).find("button.select-all").text("Unselect All");
+		}
+		else $(category).find("button.select-all").text("Select All");
+		isAllChecked = true;
 	}
 }
 
-function updateSelectAll() {
-	var tab = $(this).closest('[role="tabpanel"]');
-	var selectAll = tab.find('input.select-all');
-	var checkboxes = tab.find('.questions input[type="checkbox"]');
-	var allChecked = true;
-	checkboxes.each(function() {
-		if ( !$(this).is(':checked') )
-			allChecked = false;
-	});
-	selectAll.prop('checked', allChecked);
-}
+function bindCheckboxes() {
+	$("div.questions input:checkbox").change(uncheckLastBox);
 
-function initSelectAllBindings() {
-	var interests_box = $('.accessibility-interests');
-	var select_all_elements = interests_box.find('input.select-all');
-	var checkboxes = interests_box.find('.questions input');
-	var question_categories = interests_box.find('.category > .checkbox');
-	select_all_elements.change(selectAllChanged);
-	checkboxes.change(updateSelectAll);
-	question_categories.each(function() {
-		updateSelectAll.call(this);
-	});
+	function uncheckLastBox() {
+		var SelectAllBtn = $(this).closest("div.category").find("button.select-all");
+		var NumberOfCheckboxes = $(this).closest("div.questions").find("input:checkbox").length;
+		var NumberOfcheckedCheckboxes = $(this).closest("div.questions").find("input:checked").length;
+		var isAllChecked = (NumberOfCheckboxes == NumberOfcheckedCheckboxes);
```

Comment:
This behaves perfectly according to the requirements but a few minor changes to the code should make the code easier to read.

Please do these changes:
- Use === instead of == for the isAllChecked expression.
- Eliminate the "isSelectAllShowed" variable because it shouldn't be necessary.